docs/tdvt-test-case.md

```diff
@@ -0,0 +1,89 @@
+In this case, Tableau sent a query to the database and received data, but the tuples sent back were incorrect.
+
+Before digging too deep, consider the following cases:
+- Precision errors.
+A common problem is precision errors for floating point values, although the test framework imposes some rounding to standardize results.
+If the error is not significant then you can either skip the test or add a new expected file.
+
+- Misaligned date functions.
+Another common problem involves various date functions that might be off by a day. These can be caused by start of week errors (Sunday should be 1 and Saturday is 7). Other causes might include discrepancies around some start of week calculations or quarter boundaries. You may need to change your TDS file's connection properties to have Sunday be the first day of the week (instructions can be found [here](https://help.tableau.com/current/pro/desktop/en-us/date_properties.htm))
+
+If those cases do not apply, you'll have to look deeper at the results you got and why it isn't matching the expected values. Something that can help is to do the calculation on paper.
+
+Let's say that, for the agg.max test you are failing every one. For example, the `MAX([int0])` test case, the actual result is `1` while the expected value is `11`.
+
+If this is an expression test, the test case column will tell you the exact calculation Tableau ran. In this case, that is `MAX([int0])`. We know that this test is using the calcs table because of the TDSName column, so we can get the values of int0 from that:
+`int0 = {1, null, null, null, 7, 3, 8, null, null, 8, 4, 10, null, 4, 11, 4, 8}`
+
+Looking at the numbers, you can see that `11` is the largest one, which matches the TDVT expected value. You can also see that the lowest number is `1`, which is the actual value you got. In this contrived example, you can see that for the `MAX` function your connector is instead returning the `MIN` value.
+
+**Note:** Some calculations return multiple values. In this case, Tableau merges all duplicate values together and then sorts them.
+
+**Note:** The expected values listed in Actual Tuples are pulled from the first expected file, which may not be the closest matching expected file. For things like dates that have a lot of potential formats, this could mean your actual results look nothing like the expected. In this case, use the "Closest Expected" column in the test case and look up the correct expected file.
+
+## Case 2: Error message is "No such function...", no Generated SQL was generated and actual tuples is Null
+This error occurs if your dialect does not implement a function and Tableau tries to use it. Adding the function to the dialect will fix this issue.
+
+## Case 3: The error originated from the driver or database, Generated SQL was generated and actual tuples is Null
+If an error happened in the driver or database, you'll need to us your knowledge of the database to triage the issue.
```

Comment:
There seems to be a typo of us instead of use here.

Reply:
Fixing, thanks!